packages/prettier-plugin/test/pipes.test.ts

```diff
@@ -18,4 +18,19 @@ describe("pipes", () => {
       "pipe = 5 -> f1 -> f2 -> f3(1) -> { f: f2 }.f2 -> { foo: f3 }.foo(1)\n"
     );
   });
+  test("multi-line", async () => {
+    expect(
+      await format(
+        "pipe = 5 -> {|foobar| foobar} -> {|foobar| foobar} -> {|foobar| foobar} -> {|foobar| foobar} -> {|foobar| foobar} -> {|foobar| foobar} -> {|foobar| foobar} -> {|foobar| foobar}"
+      )
+    ).toBe(
+      `pipe = 5 -> {|foobar| foobar} -> {|foobar| foobar} -> {|foobar| foobar}
+  -> {|foobar| foobar}
+  -> {|foobar| foobar}
+  -> {|foobar| foobar}
+  -> {|foobar| foobar}
+  -> {|foobar| foobar}
```

Comment:
I'd prefer it if the result was,
```
      `pipe =
  5
  -> {|foobar| foobar}
  -> {|foobar| foobar}
  -> {|foobar| foobar}
  ...
```
like with Rescript, but couldn't figure out how to do this yet.

Reply:
Yeah, this is not easy.

For reference: https://github.com/prettier/prettier/blob/e04f0b023bda237c84280d244ee39bb45c64a44e/src/language-js/print/binaryish.js#L183

Not sure if doing something like that is necessary in our case; we already do precedence analysis, but only to decide whether to wrap things in `()` (literal characters).

To fix the indentation, we'd also have to make a decision whether to wrap things in `group(...)` (Prettier call).

We can't just move `group(...)` call to `printChild`, though, and remove it from `InfixCall` and `Pipe` branches, because sometimes the expressions that don't have `()` parens also need to be wrapped and indented.